test/unit/org/apache/cassandra/index/sai/cql/RandomIntersectionTest.java

```diff
@@ -55,23 +59,29 @@ public class RandomIntersectionTest extends SAIRandomizedTester
     public static List<Object[]> parameters()
     {
         List<Object[]> parameters = new LinkedList<>();
 
-        parameters.add(new Object[]{ "Large partition restricted high high", true, true, true, true });
-        parameters.add(new Object[]{ "Large partition restricted low low", true, true, false, false });
-        parameters.add(new Object[]{ "Large partition restricted high low", true, true, true, false });
-        parameters.add(new Object[]{ "Large partition unrestricted high high", false, true, true, true });
-        parameters.add(new Object[]{ "Large partition unrestricted low low", false, true, false, false });
-        parameters.add(new Object[]{ "Large partition unrestricted high low", false, true, true, false });
-        parameters.add(new Object[]{ "Small partition restricted high high", true, false, true, true });
-        parameters.add(new Object[]{ "Small partition restricted low low", true, false, false, false });
-        parameters.add(new Object[]{ "Small partition restricted high low", true, false, true, false });
-        parameters.add(new Object[]{ "Small partition unrestricted high high", false, false, true, true });
-        parameters.add(new Object[]{ "Small partition unrestricted low low", false, false, false, false });
-        parameters.add(new Object[]{ "Small partition unrestricted high low", false, false, true, false });
+        for (Mode mode : Mode.values())
+        {
+            parameters.add(new Object[] { "Large partition restricted, high, high, " + mode, true, true, true, true, mode });
+            parameters.add(new Object[] { "Large partition restricted, low, low, " + mode, true, true, false, false, mode });
+            parameters.add(new Object[] { "Large partition restricted, high, low, " + mode, true, true, true, false, mode });
+            parameters.add(new Object[] { "Large partition unrestricted, high, high, " + mode, false, true, true, true, mode });
+            parameters.add(new Object[] { "Large partition unrestricted, low, low, " + mode, false, true, false, false, mode });
+            parameters.add(new Object[] { "Large partition unrestricted, high, low, " + mode, false, true, true, false, mode });
+            parameters.add(new Object[] { "Small partition restricted, high, high, " + mode, true, false, true, true, mode });
+            parameters.add(new Object[] { "Small partition restricted, low, low, " + mode, true, false, false, false, mode });
+            parameters.add(new Object[] { "Small partition restricted, high, low, " + mode, true, false, true, false, mode });
+            parameters.add(new Object[] { "Small partition unrestricted, high, high, " + mode, false, false, true, true, mode });
+            parameters.add(new Object[] { "Small partition unrestricted, low, low, " + mode, false, false, false, false, mode });
+            parameters.add(new Object[] { "Small partition unrestricted, high, low, " + mode, false, false, true, false, mode });
```

Comment:
In short, the test now looks at `static AND static`, `regular AND regular`, and `regular AND static`.